plugin/storage/integration/integration.go

```diff
@@ -121,6 +125,52 @@ func (s *StorageIntegration) waitForCondition(t *testing.T, predicate func(t *te
 	return predicate(t)
 }
 
+func (s *StorageIntegration) InitArchiveStorage(t *testing.T, storageFactory storage.Factory, logger *zap.Logger) bool {
+	archiveFactory, ok := storageFactory.(storage.ArchiveFactory)
+	if !ok {
+		logger.Info("Archive storage not supported by the factory")
+		return false
+	}
+	reader, err := archiveFactory.CreateArchiveSpanReader()
+	require.NoError(t, err)
+	writer, err := archiveFactory.CreateArchiveSpanWriter()
+	require.NoError(t, err)
+	s.ArchiveSpanReader = reader
+	s.ArchiveSpanWriter = writer
+	return true
+}
+
+func (s *StorageIntegration) testArchiveTrace(t *testing.T) {
+	defer s.CleanUp()
+	success := s.InitArchiveStorage(t, s.storageFactory, zap.NewNop())
```

Comment:
None of the other testSomething tests are calling init storage. Please use the same initialization pattern.

Reply:
Done! PTAL.

Reply:
Don't see much change. You're introducing storageFactory member field that is never set, that makes all archive tests to be skipped.

Reply:
Yes while R&D, to create an ArchiveSpan Reader/Writer it needed an archiveFactory initialization that is why introduced the storageFactory member field. Had a hunch that all archive tests will be skipped due to this, wanted to run it through by you once.

Reply:
I have my college Mid-Semester exams starting from today. Can I resume working on this around next week please?